chapter-08/end/Engine/Particles/Emitter.cs

```diff
@@ -0,0 +1,107 @@
+﻿using chapter_08.Engine.Objects;
+using chapter_08.Engine.Particles.EmitterTypes;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+
+namespace chapter_08.Engine.Particles
+{
+    public class Emitter : BaseGameObject
+    {
+        private LinkedList<Particle> _activeParticles = new LinkedList<Particle>();
+        private LinkedList<Particle> _inactiveParticles = new LinkedList<Particle>();
+        private EmitterParticleState _emitterParticleState;
+        private IEmitterType _emitterType;
+        private int _nbParticleEmittedPerUpdate = 0;
+        private int _maxNbParticle = 0;
+
+        public Emitter(Texture2D texture, Vector2 position, EmitterParticleState particleState, IEmitterType emitterType, int nbParticleEmittedPerUpdate, int maxParticles)
+        {
+            _emitterParticleState = particleState;
+            _emitterType = emitterType;
+            _texture = texture;
+            _nbParticleEmittedPerUpdate = nbParticleEmittedPerUpdate;
+            _maxNbParticle = maxParticles;
+            Position = position;
+        }
+
+        public void Update(GameTime gameTime)
+        {
+            EmitParticles();
+
+            var particleNode = _activeParticles.First;
+            while (particleNode != null)
+            {
+                var nextNode = particleNode.Next;
+                var stillAlive = particleNode.Value.Update(gameTime);
+                if (!stillAlive)
+                {
+                    _activeParticles.Remove(particleNode);
+                    _inactiveParticles.AddLast(particleNode.Value);
+                }
+
+                particleNode = nextNode;
+            }
+        }
+
+        public override void Render(SpriteBatch spriteBatch)
+        {
+            var sourceRectangle = new Rectangle(0, 0, _texture.Width, _texture.Height);
+
+            foreach (var particle in _activeParticles)
+            {
+                //spriteBatch.Draw(_texture, particle.Position, Color.White);
```

Comment:
Remove this?

Reply:
I did remove it a little bit after creating the PR. It should be gone now.